stable/mysql/templates/secrets.yaml

```diff
@@ -11,16 +12,21 @@ metadata:
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
+  {{ if not .Values.allowEmptyRootPassword }}
```

Comment:
I think the check should be against `mysqlAllowEmptyPassword` for both root and non-root. If the password will actually be empty then is up to whether `mysqlPassword` or `mysqlRootPassword` is empty or not